sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +138,231 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+          .save(offering);
+
+        // Create the new assessment to be processed.
+        scholasticStanding.studentAssessment = {
+          application: { id: application.id } as Application,
+          triggerType: AssessmentTriggerType.ScholasticStandingChange,
+          creator: auditUser,
+          submittedBy: auditUser,
+          submittedDate: now,
+          offering: { id: savedOffering.id } as EducationProgramOffering,
+        } as StudentAssessment;
+      } else {
+        // If unsuccessful weeks, then add to the column.
+        // * No cloning of offering and re-assessment is required in this scenario.
+        scholasticStanding.unsuccessfulWeeks =
+          scholasticStandingData.numberOfUnsuccessfulWeeks;
+      }
 
       const studentScholasticStanding = await transactionalEntityManager
         .getRepository(StudentScholasticStanding)
         .save(scholasticStanding);
 
-      // Save current application.
-      application.currentAssessment = {
-        id: scholasticStanding.studentAssessment.id,
-      } as StudentAssessment;
+      if (scholasticStanding.studentAssessment) {
+        // Save current assessment to application, if any.
+        application.currentAssessment = {
+          id: scholasticStanding.studentAssessment.id,
+        } as StudentAssessment;
+      }
 
       // Set archive to true
       application.isArchived = true;
 
+      // Save current application.
       await transactionalEntityManager
         .getRepository(Application)
         .save(application);
 
       return studentScholasticStanding;
     });
   }
+
+  /**
+   * The service process the payload data and checks for certain restriction,
+   * and add new restrictions, if required.
+   * When institution report withdrawal for a FT course application,
+   * add WTHD restriction to student.
+   * When institution report Withdrawal for a FT course on a student WITH a WTHD
+   * restriction add SSR restriction.
+   * When institution reports a change related to a FT application for unsuccessful
+   * completion and the total number of unsuccessful weeks hits minimum 68, add SSR
+   * restriction.
+   * When institution report Withdrawal OR unsuccessful for a PT course application,
+   * add PTSSR restriction to student.
+   * If a ministry user resolves the SSR or PTSSR or WTHD restriction, and new withdrawal
+   * is reported, re add the above restrictions.
+   * If a ministry user resolves the SSR restriction, and new unsuccessful completion
+   * is reported, add the restriction (minimum is still at least 68).
+   * @param scholasticStandingData scholastic standing data.
+   * @param offeringIntensity offering intensity.
+   * @param studentId student id.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   * @param applicationId application id.
+   * @returns a new student restriction object, that need to be saved.
+   */
+  async manageStudentRestrictions(
```

Comment:
IMO this method is not `managing` something it is just defining if a restriction must be created.
IMO it can be renamed to `getScholasticStandingRestriction`.